src/DependencyManagement/DependencyManagementUtils.cs

```diff
@@ -0,0 +1,114 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.IO;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Functions.PowerShellWorker.DependencyManagement
+{
+    internal class DependencyManagementUtils
+    {
+        // File name which holds module information after it is downloaded/installed from the PSGallery.
+        private const string DependencyInfoJsonFileName = "DependencyInfo.json";

```

Comment:
What is the purpose of DependencyInfo.json? when you are already building denpendencyinfo objects

Reply:
@pragnagopa : When we install a module (via Save-Module), we create a DependecyInfo.json entry with the name and version of the module that was installed. This way, next time the function app is initialized, we can use this file to try to figure out if we have already installed the module at the requested path. If the module is already there, we check the version to ensure we have the latest one. For reference, please see RunInstallManagedDependencyModule

Reply:
Can you expand on what you mean by 

> next time the function app is initialized

We can sync up in person if needed.

Reply:
@pragnagopa : Let's sync in person tomorrow. It would be easier to explain the logic in the whiteboard.